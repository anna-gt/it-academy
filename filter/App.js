import React from 'react';
import ReactDOM from 'react-dom';
import Filter from './components/Filter';


import words from './words.json';

ReactDOM.render(
  <Filter
	words = {words}
  />
  , document.getElementById('container') 
);
