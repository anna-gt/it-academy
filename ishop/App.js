import React from 'react';
import ReactDOM from 'react-dom';

import IShop from './components/IShop';

const iShopName='Товары для питомцев';
const names = 'Наименование';
const images = 'Изображение';
const prices = 'Цена, р';
const qts = 'Остаток, шт';

import itemsArr from './items.json';

ReactDOM.render(
  <IShop 
    name = {iShopName}
    items = {itemsArr}
    names = {names}
    images = {images}
    prices = {prices}
    qts = {qts}
  />
  , document.getElementById('container') 
);
