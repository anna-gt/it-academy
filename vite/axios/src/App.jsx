import React, { useState, useEffect } from 'react'
import axios from 'axios'
import UserData from './components/UserData'
import OnLoadingUserData from './components/OnLoadingUserData'
import './App.css'

function App() {

	const [appState, setAppState] = useState(
		{
      loading: false,
      persons: null,
    }
	);
	const DataLoading =  OnLoadingUserData(UserData);

	useEffect(() => {
		const apiUrl='http://www.filltext.com/?rows=32&id=%7Bnumber%7C1000%7D&firstName=%7BfirstName%7D&lastName=%7BlastName%7D&email=%7Bemail%7D&phone=%7Bphone%7C(xxx)xxx-xx-xx%7D&address=%7BaddressObject%7D&description=%7Blorem%7C32%7D';
		axios.get(apiUrl).then((response) => {
			const allPersons = response.data;
			setAppState(
				{
					loading: false,
					persons: allPersons
					 }
			);
		})

	}, [setAppState])

  return (
    <>
			<div className="app">
        <DataLoading isLoading={appState.loading} persons={appState.persons} />
    </div>
    </>
  )
}

export default App
