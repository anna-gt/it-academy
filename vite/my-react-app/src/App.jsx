import { useState, useEffect } from 'react'
import './App.css'
import {BrowserRouter, Routes, Route, Link} from "react-router-dom";

// function App() {
// 	const [data, setData] = useState(null);
// 	const [toggle, setToggle] = useState(false);
// 	const [value, setValue] = useState('');

// 	const onClick = () => setToggle(prev => !prev);

// 	useEffect(() => {
// 		setTimeout(() => {
// 			setData({})
// 		}, 100)
// 	}, [])

//   return (
//     <>
// 			<h1 data-testid='value-elem'>{value}</h1>
// 			{toggle === true && <div data-testid='toggle-elem'>toggle</div>}
// 			{data && <div>data</div>}
//       Hello world
// 			<button data-testid='toggle-btn' onClick={onClick}>click me</button>
// 			<input onChange={eo => setValue(eo.target.value)} type='text' placeholder='input value...'></input>
//     </>
//   )
// }


import AppRouter from './router/AppRouter';
import Navbar from './components/Navbar/Navbar';

const App = () => {
	return (
		<div>
			<Navbar />
			<AppRouter />
		</div>
	)
}

export default App
