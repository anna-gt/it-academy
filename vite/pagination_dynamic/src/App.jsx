import { useState, useEffect } from 'react'
import axios from 'axios'
import './App.css'


function App() {
	const [photos, setPhotos] = useState([]);
	const [currentPage, setCurrentPage] = useState(1);
	const [fetching, setFetching] = useState(true);
	const [totalCount, setTotalCount] = useState(0);

	useEffect(() => {
		if (fetching) {
			console.log('fetching');
			axios.get(`https://jsonplaceholder.typicode.com/photos?_limit=10&_page=${currentPage}`)
				.then(response => {
					setPhotos([...photos,...response.data]);
					setCurrentPage(prevState => prevState + 1);
					setTotalCount(response.headers['x-total-count'])
				})
				.catch(error => console.log('error'))
				.finally(() => setFetching(false))
		}
	}, [fetching])


	useEffect(() =>{
		document.addEventListener('scroll', scrollHandler)
		return function () {
			document.removeEventListener('scroll', scrollHandler)
		}
	}, [totalCount]);

	const scrollHandler = (eo) => {
		if (eo.target.documentElement.scrollHeight - (eo.target.documentElement.scrollTop + window.innerHeight) < 100 
		&& photos.length < totalCount) {
			setFetching(true);
		}
	};

	const photoCode = photos.map(photo =>
		<div className='photo' key={photo.id}>
			<div className='title'>{photo.id}. {photo.title}</div>
			<img src={photo.thumbnailUrl} alt=""/>
		</div>)

  return (
    <div className='app'>
			{photoCode}
    </div>
  )
}

export default App
