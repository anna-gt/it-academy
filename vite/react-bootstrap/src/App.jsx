import { useState } from 'react'
import './App.css'
import {Button, Alert, Breadcrumb} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

function App() {

  return (
    <>
		<Breadcrumb>
      <Breadcrumb.Item href="#">Home</Breadcrumb.Item>
      <Breadcrumb.Item href="https://getbootstrap.com/docs/4.0/components/breadcrumb/">
        Library
      </Breadcrumb.Item>
      <Breadcrumb.Item active>Data</Breadcrumb.Item>
    </Breadcrumb>
		<Alert>This is a button</Alert>
    <Button>TEST BUTTON</Button>
    </>
  )
}

export default App
