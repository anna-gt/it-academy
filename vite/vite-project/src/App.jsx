import { useState, useEffect } from 'react';
import useSWR from "swr";
import './App.css'

async function dataFetcher() {
	const response = await fetch('https://jsonplaceholder.typicode.com/photos?_limit=10&_page=3', {
		method: 'get',
		headers: {
      "Accept": "application/json",
    },
	});
	if (!response.ok) {
		throw new Error("fetch error " + response.status);
	}
	return response.json();
}

function App() {
	const [photos, setPhotos] = useState([]);

	const { data, error, isLoading } = useSWR(
		"jsonphotos",
		dataFetcher
	);

	useEffect(() => {
		setPhotos(data);
		console.log(data)
	},[data])


	useEffect(() =>{
		document.addEventListener('scroll', scrollHandler)
		return function () {
			document.removeEventListener('scroll', scrollHandler)
		}
	}, []);

	const scrollHandler = (eo) => {
		console.log(scroll);
	};

	if ( error ) {
    return <div>Ошибка: {error}</div>;
  }

  if ( isLoading ) {
    return <div>загрузка данных...</div>;
  }

	const photoCode = photos.map(photo =>
		<div className='photo' key={photo.id}>
			<div className='title'>{photo.id}. {photo.title}</div>
			<img src={photo.thumbnailUrl} alt=""/>
		</div>)

  return (
    <div className='app'>
			{photoCode}
    </div>
  )
}

export default App
