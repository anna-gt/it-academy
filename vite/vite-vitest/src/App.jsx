import { useState } from 'react'
import './App.css'
import React from 'react';
import {BrowserRouter, Routes, Route, Link} from "react-router-dom";
import MainPage from "./pages/MainPage";
import AboutPage from "./pages/AboutPage";
import Users from './users/Users';
import UserDetailsPage from './pages/UserDetailsPage';
import ErrorPage from './pages/ErrorPage';

function App() {

  return (
    <>
      <BrowserRouter>
				<h1>Hello world</h1>
				<Link to="/" data-testid="main-link">main</Link>
				<Link to="/about" data-testid="about-link">about</Link>
				<Link to="/users" data-testid="users-link">users</Link>
					<Routes>
						<Route path="/" element={<MainPage/>} />
						<Route path="/about" element={<AboutPage/>} />
						<Route path="/users" element={<Users/>} />
						<Route path="/users/:id" element={<UserDetailsPage/>} />
						<Route path="/*" element={<ErrorPage/>} />
					</Routes>
			</BrowserRouter>
    </>
  )
}

export default App
