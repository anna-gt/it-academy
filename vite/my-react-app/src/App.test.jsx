import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import App from './App';
import '@testing-library/jest-dom';

describe('TEST APP', () => {
	// test('renders learn vite ', () => {
	// 	render(<App />);
	// 	const HelloElem = screen.getByText(/hello world/i);
	// 	const btn = screen.getByRole('button');
	// 	const input = screen.getByPlaceholderText(/input value/i);
	// 	expect(HelloElem).toBeInTheDocument();
	// 	expect(btn).toBeInTheDocument();
	// 	expect(input).toMatchSnapshot();
	// });

	// test('renders learn vite ', async () => {
	// 	render(<App />);
	// 	// const helloElem = screen.queryByText(/hello2/i)
	// 	// expect(helloElem).toBeNull()
	// 	const helloWorldElem = await screen.findByText(/data/i)
	// 	expect(helloWorldElem).toBeInTheDocument();
		
	// });

	// test('click event', () => {
	// 	render(<App />);
	// 	const btn = screen.getByTestId('toggle-btn');
	// 	expect(screen.queryByTestId('toggle-elem')).toBeNull();
	// 	fireEvent.click(btn)
	// 	expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
	// 	fireEvent.click(btn)
	// 	expect(screen.queryByTestId('toggle-elem')).toBeNull();
	// });

	// test('input event', async () => {
	// 	render(<App />);
	// 	const input = screen.getByPlaceholderText(/input value/i);
	// 	expect(screen.queryByTestId('value-elem')).toContainHTML('');
	// 	screen.debug();
	// 	// Искуственное событие
	// 	// fireEvent.input(input, {
	// 	// 	target: {value: '131455'}
	// 	// })
	// 	// Близко к пользователю, обрабатываются события нажатия клавиш и тд
	// 	const user = userEvent.setup()
	// 	await user.type(input, '131455' )
	// 	expect(screen.getByTestId('value-elem')).toContainHTML('131455');
	// 	screen.debug();
	// });
	
})

