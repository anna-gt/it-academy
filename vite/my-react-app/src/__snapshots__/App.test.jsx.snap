// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP renders learn vite  1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
